build.gradle

```diff
@@ -55,7 +55,7 @@ configurations {
         exclude group: 'org.slf4j', module: 'slf4j-api'
     }
 
-    compileTestKotlin {
+    tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).all {
```

Comment:
Can you please explain the reason to change this compileTestKotlin to tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).all

What is the difference it make by changing this.

Reply:
tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).all ensures that the JVM target is set to 17 for all Kotlin compilation tasks, including both main and test sources. while compileTestKotlin only sets the JVM target to 17 for Kotlin code in the test sources.

Reply:
@aparnamichael Thank you for asking this question, I was wondering about this myself.

@anusreelakshmi934 For clarification, is the error you mention here the same as the `NoSuchMethodException` mentioned in your earlier comments?

 Also, when looking over the StackOverflow article provided, I noticed it was focused on Android projects and the answer had this line:
 "_There's no compileKotlin task, which is usually created for the main source set in ordinary Java + Kotlin projects._"
 I believe our project would be an "ordinary Java + Kotlin" project. Did you try using `compileKotlin` on this line?

Reply:
Initially, I changed the version of `'org.jetbrains.intellij'` to `'1.17.3'`, which led to an error stating "Unsupported class file major version 63". Investigating upon that, I found a solution on [this GitHub thread](https://github.com/react-native-community/cli/issues/1719), suggesting to modify the `distributionUrl` in the 'gradle-wrapper.properties' file. I opted for `gradle-7.6-all.zip`, but this change resulted in a `NoSuchMethodException.`

Further exploring, I updated the version to the latest `gradle-8.7-all.zip` . Upon changing that i got an error - `Could not find method kotlinOptions() for arguments [build_7jqpmvy23lx9badxvjae0ix41$_run_closure2$_closure22$_closure23@13223cc3] on configuration ':compileTestKotlin' of type org.gradle.api.internal.artifacts.configurations.DefaultUnlockedConfiguration`. To resolve this, I replaced `compileTestKotlin` with `tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).all.` Now everything works fine without any exceptions or errors. 
I also attempted using compileKotlin, but faced the same aforementioned error.

Reply:
@anusreelakshmi934 thanks for the explanation. Please add a comment explaining what this line is doing.

Reply:
Added the explanation